icu4j/main/core/src/main/java/com/ibm/icu/text/NFRule.java

```diff
@@ -131,6 +131,10 @@ public static void makeRules(
             StringBuilder sbuf = new StringBuilder();
             int orElseOp = description.indexOf('|');
 
+            long mod = power(rule1.radix, rule1.exponent);
```

Comment:
Please mirror the ICU4C change. The power function should be called once instead of twice.